packages/compass-shell/src/components/compass-shell/compass-shell.jsx

```diff
@@ -7,29 +7,74 @@ import { connect } from 'react-redux';
 import { Shell } from 'mongosh-browser-repl';
 
 export class CompassShell extends Component {
-  static displayName = 'CompassShellComponent';
-
   static propTypes = {
-    runtime: PropTypes.object
+    runtime: PropTypes.object,
+    historyStorage: PropTypes.object
   };
 
   static defaultProps = {
     runtime: null
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      initialHistory: this.props.historyStorage ? null : []
+    };
+  }
+
+  componentDidMount() {
+    this.loadHistory();
+  }
+
+  saveHistory = async(history) => {
+    if (!this.props.historyStorage) {
+      return;
+    }
+
+    try {
+      await this.props.historyStorage.save(history);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error);
+    }
+  }
+
+  loadHistory = async() => {
```

Comment:
what if you load up history when this plugin is loaded? (`onActivated`) that way your props always have history and then you add and remove it as you go.

Reply:
^ reason I am saying this is it might be nice to have this logic out of a view component and in a redux state management part of the plugin.

Reply:
That's good too, had a chat with rhys and decided to avoid redux as there is nothing else interested in that state, but i'm open to changing it.

Where the save history goes in that case? We could also set two properties: `initialHistory` and `saveHistory`.

Reply:
yea using redux state will definitely require you to change a few things so up to you!

remind me of the difference between saveHistory and initialHistory?

Reply:
`saveHistory` would be the function called to save the history when it changes `initialHistory` would be the array of history entries you suggested

Reply:
I think i'll merge this as is, and revisit it now that we integrate the shell properly, as i believe that is going to involve more redux and it will make sense to have it consistent